packages/hub/tasks/create-profile.ts

```diff
@@ -0,0 +1,36 @@
+import { inject } from '@cardstack/di';
+import { query } from '@cardstack/hub/queries';
+import { service } from '@cardstack/hub/services';
+import { encodeDID } from '@cardstack/did-resolver';
+import * as Sentry from '@sentry/node';
+
+export default class CreateProfile {
+  jobTickets = query('job-tickets', { as: 'jobTickets' });
+  merchantInfoQueries = query('merchant-info', {
+    as: 'merchantInfoQueries',
+  });
+  relay = service('relay');
+  workerClient = inject('worker-client', { as: 'workerClient' });
+
+  async perform({
+    'job-ticket-id': jobTicketId,
+    'merchant-info-id': merchantInfoId,
+  }: {
+    'job-ticket-id': string;
+    'merchant-info-id': string;
+  }) {
+    try {
+      let merchantInfo = (await this.merchantInfoQueries.fetch({ id: merchantInfoId }))[0];
+      let did = encodeDID({ type: 'MerchantInfo', uniqueId: merchantInfoId });
+
+      let merchantSafeAddress = await this.relay.registerProfile(merchantInfo.ownerAddress, did);
```

Comment:
I think the relayer will return the transaction id and we need to wait here for the transaction to be finished or failed, and update the job ticket accordingly.

Reply:
ah okay, does that mean subgraph support will be needed?

Reply:
good point, the relayer is not indexing V2 market contract txns. currently the SDK using the subgraph to look for txn hashes so that there is consistency between what is on chain and what is in the subgraph. not using the subgraph would force clients of this to have to use web3 to monitor txn completion instead of the SDK

Reply:
@habdelra we're not dealing with the `PrepaidCardMarketV2` contract here, just the `MerchantManager` (we use the `registerMerchant` function). It looks like `MerchantCreation` event is indexed in the subgraph so I think we can use the SDK for waiting for the transaction to finish.

Reply:
It looks like `waitForTransactionConsistency` [isn’t exported](https://github.com/cardstack/cardstack/blob/69dd3bad2341981cdcf9afb12445940ef49f29ff/packages/cardpay-sdk/index.ts#L38), is there a higher-level function I should be using? Or should I add an export for it?

I tried that locally but I think I’d need to publish the SDK for it to work…?

Reply:
okay, I was finally able to use that function locally, I tried it with [this prepaid card issuance transaction](https://blockscout.com/xdai/mainnet/tx/0xf47180533b1eada183ec94b29db7dc2707041b88607a1fca4ac21d27ca23fd82/internal-transactions):

```
result = await waitForTransactionConsistency(
  window['@cardstack/WebClient'].__container__.lookup('service:layer2-network').strategy.web3,
  '0xf47180533b1eada183ec94b29db7dc2707041b88607a1fca4ac21d27ca23fd82')
```

I didn’t see the address of the prepaid card (`0x362b921cFe6a72731219AE9366784DF742cE06f9`) in the result though

What would it look like to extract the merchant safe address from the result of this function?